torch/distributed/fsdp/_init_utils.py

```diff
@@ -395,13 +395,19 @@ def _init_param_handle_from_module(
     """
     _check_single_device_module(fully_sharded_module, state._ignored_params)
     device_from_device_id = _get_device_from_device_id(device_id, state.rank)
-    _materialize_module(
-        fully_sharded_module,
-        param_init_fn,
-        state._ignored_params,
-        device_from_device_id,
-        lambda k: not isinstance(k, module_wrapper_cls),
+    is_meta_module, is_torchdistX_deferred_init = _need_to_materialize_module(
+        fully_sharded_module, state._ignored_params
     )
+    # Materialize the module if needed
+    if (is_meta_module or is_torchdistX_deferred_init) and param_init_fn is not None:
```

Comment:
I thought if param_init_fn is None, but we have meta modules / params, we should try to do some default initialization for the user?